src/mnemonic.rs

```diff
@@ -0,0 +1,139 @@
+use std::fmt;
```

Comment:
This whole file is duplicating quite a bit of code to be able to inject a new implementation of the `parse_share` function. I think it would make more sense to remove this function from the `Shamir` trait, and treat parsing separately in the `main.rs` file directly (potentially with a separate parsing trait). In hindsight, parsing shares is covered by the `Shamir` trair but parsing secrets is not, which is imbalanced.